src/mam4xx/ndrop.hpp

```diff
@@ -22,87 +22,110 @@ class NDrop {
+           // from previous step at level k-1 *** this should only be present if
+           // the current species is unactivated number/sfc/mass
+    const Real qactold_kp1 =
+        1 // optional: number / mass mixing ratio of ACTIVATED species
+          // from previous step at level k+1 *** this should only be present if
+          // the current species is unactivated number/sfc/mass
 ) {
 
-  int top_lev = 0;
-
-  Kokkos::parallel_for(
-      Kokkos::TeamThreadRange(team, nlev), KOKKOS_LAMBDA(int k) {
-        int kp1 = min(k + 1, nlev - 1);
-        int km1 = max(k - 1, top_lev);
-
-        // the qactold*(1-overlap) terms are resuspension of activated material
-
-        if (is_unact) {
-          q(k) = qold(k) +
-                 (dt * (-src(k) +
-                        (ekkp(k) * (qold(kp1) - qold(k) +
-                                    (qactold(kp1) * (1 - overlapp(k))))) +
-                        (ekkm(k) * (qold(km1) - qold(k) +
-                                    (qactold(km1) * (1 - overlapm(k)))))));
-        } else {
-          q(k) = qold(k) +
-                 (dt *
-                  (src(k) + (ekkp(k) * ((overlapp(k) * qold(kp1)) - qold(k))) +
-                   (ekkm(k) * ((overlapm(k) * qold(k)) - qold(k)))));
-        }
-        // force to non-negative
-        q(k) = max(q(k), 0);
-      });
-}
+  // the qactold*(1-overlap) terms are resuspension of activated material
+
+  if (is_unact) {
+    q = qold_k +
+        (dt *
+         (-src +
+          (ek_kp1 * (qold_kp1 - qold_k + (qactold_kp1 * (1 - overlap_kp1)))) +
+          (ek_km1 * (qold_km1 - qold_k + (qactold_km1 * (1 - overlap_km1))))));
+  } else {
+    q = qold_k + (dt * (src + (ek_kp1 * ((overlap_kp1 * qold_kp1) - qold_k)) +
+                        (ek_km1 * ((overlap_km1 * qold_k) - qold_k))));
+  }
+  // force to non-negative
+  q = max(q, 0);
+} // end explmix
+
+// calculates maximum supersaturation for multiple
+// competing aerosol modes.
+// Abdul-Razzak and Ghan, A parameterization of aerosol activation.
+// 2. Multiple aerosol types. J. Geophys. Res., 105, 6837-6844.
+KOKKOS_INLINE_FUNCTION
+void maxsat(const Real zeta,   // [dimensionless]
+            const Real eta[4], // [dimensionless]
+            const Real nmode,  // modes
+            const Real smc[4], // critical supersaturation for number mode
+                               // radius [fraction]
+            Real &smax         // maximum supersaturation [fraction] (output)
+) {
+  // abdul-razzak functions of width
+  Real f1[AeroConfig::num_modes()];
+  Real f2[AeroConfig::num_modes()];
+
+  Real sum = 0;
+  Real g1, g2;
+  bool weak_forcing = true; // whether forcing is sufficiently weak or not
+
+  for (int m = 0; m < nmode; m++) {
+    if (zeta > 1e5 /*FIXME: BAD CONSTANT*/ * eta[m] ||
+        smc[m] * smc[m] > 1e5 /*FIXME: BAD CONSTANT*/ * eta[m]) {
+      // weak forcing. essentially none activated
+      smax = 1e-20; /*FIXME: BAD CONSTANT*/
```

Comment:
We could define a variable for this  numerical constant, i.e., `Real const small = 1e-20;`

Reply:
This constant could go at the top of the function.